openstudiocore/src/energyplus/ReverseTranslator/ReverseTranslateZonePropertyUserViewFactorsBySurfaceName.cpp

```diff
@@ -0,0 +1,140 @@
+    return boost::none;
+  }
+
+  boost::optional<WorkspaceObject> target = workspaceObject.getTarget(ZoneProperty_UserViewFactors_bySurfaceNameFields::ZoneorZoneListName);
+
+  std::vector<ThermalZone> thermalZones;
+
+  if (target){
+
+    boost::optional<ModelObject> mo;
+    if (target->iddObject().type() == IddObjectType::Zone){
+      mo = translateAndMapWorkspaceObject(target.get());
+      if( mo ) {
+        if( boost::optional<Space> space = mo->optionalCast<Space>() ) {
+          boost::optional<ThermalZone> thermalZone = space->thermalZone();
+          if (thermalZone){
+            thermalZones.push_back(*thermalZone);
+          }
+        }
+      }
+    }
+  }
+
+  if(thermalZones.empty())
+  {
+    LOG(Error, "Error importing object: "
+             << workspaceObject.briefDescription()
+             << " Can't find associated ThermalZone.");
+
+    return boost::none;
+  }
+
+  boost::optional<ModelObject> result;
+  for (ThermalZone thermalZone : thermalZones){
+
+    openstudio::model::ZonePropertyUserViewFactorsBySurfaceName zoneProp = thermalZone.getZonePropertyUserViewFactorsBySurfaceName();
+
+    if (!result){
+      result = zoneProp;
+    }
+
+    for (const IdfExtensibleGroup& idfGroup : workspaceObject.extensibleGroups()){
+      WorkspaceExtensibleGroup workspaceGroup = idfGroup.cast<WorkspaceExtensibleGroup>();
+
+      boost::optional<WorkspaceObject> fromTarget = workspaceGroup.getTarget(ZoneProperty_UserViewFactors_bySurfaceNameExtensibleFields::FromSurface);
+      boost::optional<WorkspaceObject> toTarget = workspaceGroup.getTarget(ZoneProperty_UserViewFactors_bySurfaceNameExtensibleFields::ToSurface);
+
+      boost::optional<ModelObject> toModelObject;
+      boost::optional<ModelObject> fromModelObject = translateAndMapWorkspaceObject(*fromTarget);
+      if (fromTarget != toTarget){
+        toModelObject = translateAndMapWorkspaceObject(*toTarget);
+      } else {
+        toModelObject = fromModelObject;
+      }
+
+      OptionalDouble _viewFactor = workspaceGroup.getDouble(ZoneProperty_UserViewFactors_bySurfaceNameExtensibleFields::ViewFactor);
+
+      // If the objects translated ok, and the viewFactor double value exists
+      if (fromModelObject && toModelObject && _viewFactor) {
+        // try to create a viewFactor, which will throw in case wrong type, or wrong double value. Here we let it slide though
+        try {
+          ViewFactor viewFactor(fromModelObject.get(), toModelObject.get(), _viewFactor.get());
+          // add the view factor
+          if (zoneProp.addViewFactor(viewFactor)) {
+            LOG(Warn, "Adding ViewFactor in ThermalZone " << thermalZone.nameString()
+                  << " failed for viewFactor=" << viewFactor << ".");
+          }
+        } catch (...) {
+          // The ViewFactor Ctor threw, so there's a wrong type, or a wrong double
+          LOG(Error, "Couldn't not create ViewFactor in ThermalZone " << thermalZone.nameString()
```

Comment:
Remove "not"?

Reply:
No Keep it here

Reply:
Should either be "Couldn't create" or "Could not create"

Reply:
Yes, that's what I meant.

Reply:
Yes, read it too quickly (just like I wrote it too quickly...). This should not be a double negation, just express that it was not possible. Another brick in the wall:

> We don't need no education...